datadog/java_agent.go

```diff
@@ -25,6 +25,7 @@ type JavaAgent struct {
 
 func NewJavaAgent(dependency libpak.BuildpackDependency, cache libpak.DependencyCache, logger bard.Logger) JavaAgent {
 	contrib, _ := libpak.NewDependencyLayer(dependency, cache, libcnb.LayerTypes{
+		Build: true,
```

Comment:
I would suggest you pass in something as an argument here, like `nativeImage bool` or `enableAtBuild bool` and use that set the build flag. You can also store that on the struct and use it during `Contribute` to conditionally add the build environment arguments.

You'll need to pass this value in from build.go, but you can  basically pass in `cr.ResolveBool("BP_NATIVE_IMAGE")`.

Reply:
Thanks for the feedback. I tried my best from b61a01d6cc1c563ad146c9650c039d9510dca5bc by:
- Getting a `ConfigurationResolver` from `build.go` and checking `BP_NATIVE_IMAGE` to create a flag,
- Passing this flag to the `JavaAgent` structure and to the new layer `Build` flag,
- Applying `native-image` compiler option only if the `JavaAgent` flag is set.

Please let me know if this is what you had in mind?

Reply:
Yes, that looks good. Thanks